receiver/collectdreceiver/config_test.go

```diff
@@ -53,7 +57,11 @@ func TestLoadConfig(t *testing.T) {
 			require.NoError(t, err)
 			require.NoError(t, component.UnmarshalConfig(sub, cfg))
 
-			assert.NoError(t, component.ValidateConfig(cfg))
+			if tt.wantErr == nil {
+				assert.NoError(t, component.ValidateConfig(cfg))
+			} else {
+				assert.Equal(t, tt.wantErr, component.ValidateConfig(cfg))
+			}
```

Comment:
You should be able to use `assert.ErrorIs` here and avoid the if statement :)

Reply:
I tried just now, it didn't seem to work because the error on validation is built with `fmt.Errorf` and deep equality fails or something.